src/main/java/org/gridsuite/loadflow/server/service/LoadFlowWorkerService.java

```diff
@@ -248,6 +280,211 @@ protected List<LimitViolationInfos> calculateOverloadLimitViolations(List<LimitV
+
+        // load computation by country
+        network.getLoads().forEach(load -> {
+            Terminal terminal = load.getTerminal();
+            Pair<Country, Double> countryAndPFromLoad = getCountryAndPFromTerminalInMainComponent(terminal);
+            Country country = countryAndPFromLoad.getLeft();
+            if (country != null) {
+                double p = countryAndPFromLoad.getRight();
+                fillCountryAdequacy(adequaciesByCountry, country, CountryAdequacy.ValueType.LOAD, p);
+            }
+        });
+
+        // generation computation by country
+        Stream.concat(network.getGeneratorStream(), network.getBatteryStream()).forEach(injection -> {
+            Terminal terminal = injection.getTerminal();
+            Pair<Country, Double> countryAndPFromInjection = getCountryAndPFromTerminalInMainComponent(terminal);
+            Country country = countryAndPFromInjection.getLeft();
+            if (country != null) {
+                double p = countryAndPFromInjection.getRight();
+                fillCountryAdequacy(adequaciesByCountry, country, CountryAdequacy.ValueType.GENERATION, abs(p));
+            }
+        });
+
+        // net position computation by country
+        borderBasedCountryAreas.forEach((country, borderBasedCountryArea) -> {
+            double netPosition = borderBasedCountryArea.getNetPosition();
+            fillCountryAdequacy(adequaciesByCountry, country, CountryAdequacy.ValueType.NET_POSITION, netPosition);
+        });
+
+        // losses computation by country : P - C - net position
+        adequaciesByCountry.forEach((key, value) ->
+            value.setLosses(value.getGeneration() - value.getLoad() - value.getNetPosition())
+        );
+
+        return adequaciesByCountry.entrySet().stream()
+            .map(entry -> CountryAdequacy.builder()
+                .country(entry.getKey())
+                .load(entry.getValue().getLoad())
+                .generation(entry.getValue().getGeneration())
+                .losses(entry.getValue().getLosses())
+                .netPosition(entry.getValue().getNetPosition())
+                .build())
+            .collect(Collectors.toList());
+    }
+
+    private void fillExchange(Map<String, List<Exchange>> result, String country, String otherCountry, double exchange) {
+        List<Exchange> exchangesCountrytoOtherCountries = result.computeIfAbsent(country, k -> new ArrayList<>());
+
+        OptionalInt indexOfOtherCountry = IntStream.range(0, exchangesCountrytoOtherCountries.size())
+            .filter(i -> exchangesCountrytoOtherCountries.get(i).getCountry().equals(otherCountry))
+            .findFirst();
+        if (indexOfOtherCountry.isEmpty()) {
+            exchangesCountrytoOtherCountries.add(new Exchange(null, otherCountry, exchange));
+        }
+    }
+
+    private Pair<Country, Double> getCountryAndPFromTerminalInMainComponent(Terminal terminal) {
```

Comment:
what does PF stands for ?
maybe rename the function to be explicit ?

Reply:
:+1: